### PR TITLE
Sarif genesis

### PR DESCRIPTION
**Pull request by tarunKoyalwar**
# Description

- Simple and Standard Implementation of SARIF for nuclei 
- Supports both Import & Export

- Create an example report by running

```sh
  go run example/main.go
```

It can be viewed using VSCode Sarif Extension or at [sarif-webcomponent](https://microsoft.github.io/sarif-web-component/)

Screenshot 2022-11-26 at 1 30 46 AM

### CODE REVIEW

**Comment by tarunKoyalwar** [comment]
Implemented for https://github.com/projectdiscovery/nuclei/issues/2608

**Comment by tarunKoyalwar** [comment]
With Implementation of Sarif, output of [nuclei-gh-action](https://github.com/projectdiscovery/nuclei-action) can be viewed in Security/Code-Scanning-Results of that repository.

as requested at https://github.com/projectdiscovery/nuclei-action/issues/43

something similar to https://www.youtube.com/watch?v=4MduRFoKpGY

File: example/main.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+import (
+	"log"
+	"os"
+
+	"github.com/projectdiscovery/sarif"
+)
+
+func main() {
+
```
Comment (Mzack9999) [inline example/main.go]: extra line

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module github.com/projectdiscovery/sarif
+
+go 1.19
```
Comment (Mzack9999) [inline go.mod]: we still target go 1.18

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
lgtm - minor changes requested

**Review by Mzack9999 — APPROVED** [review]

**Pull request merged**